bigchaindb/common/utils.py

```diff
@@ -46,3 +49,17 @@ def deserialize(data):
             string.
     """
     return rapidjson.loads(data)
+
+
+def validate_asset_data_keys(tx_body):
+    backend = bigchaindb.config['database']['backend']
+
+    if backend == 'mongodb':
+        data = tx_body['asset'].get('data', {})
+        keys = data.keys() if data else []
```

Comment:
This is only going to extract the top-level keys in `asset.data` but I suspect the restriction on key names is true for keys at all levels in the hierarchy / tree.